### PR TITLE
Review terraform

### PR DESCRIPTION
**Pull request by alexronquillo**
## Description

Language edits of old Terraform/Terragrunt docs

### CODE REVIEW

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 1]
```diff
@@ -304,33 +338,45 @@ module "HostConditions" {
 }
 ```
 
-You have added a New Relic alert channel, subscribed the demo policy to the alert channel, and added a module hosted on Github.
-
-To clone the module from GitHub, run `terragrunt init` and then run `terragrunt apply`. After Terraform is done processing, You will see your Demo Policy now has two conditions in your New Relic account and one alert channel attached.
+You've added a New Relic alert channel, subscribed the demo policy to the alert channel, and added a module hosted on Github.
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:341]: Here, you added a New Relic alert channel, subscribed the demo policy to the alert channel, and added a module hosted on Github.

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 2]
```diff
@@ -255,24 +273,40 @@ Now, when you run `terragrunt plan`, Terragrunt will provide the input block's v
...
+If you haven't done these guides yet, you can copy their configurations from the [Terragrunt intro Github repo](https://github.com/jsbnr/nr-terragrunt-intro-example).
 
 </Callout>
 
-Update the __main.tf__ to include the additional terraform configurations. Update the email address in the alert channel to your preferred email address:
+<Steps>
+
+<Step>
+
+In _main.tf_, update the email address in the alert channel to your preferred email address:
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:308]: _main.tf_ is inside src folder. I suggested adding this reference for clarification.

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 3]
```diff
@@ -304,33 +338,45 @@ module "HostConditions" {
...
+<Step>
+
+Run `terragrunt init` and then run `terragrunt apply`:
+
+```bash
+terragrunt init
+terragrunt apply
+```
+
+After Terraform finishes processing, your alert policy has two conditions in your and one alert channel.
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:354]:
```suggestion
After Terraform finishes processing, your alert policy has two conditions and one alert channel.
```

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 4]
```diff
@@ -304,33 +338,45 @@ module "HostConditions" {
...
 
-In the root __terragrunt.hcl__ file, create a new input for `env_name`
 ```hcl
 inputs = {
 	env_name = "develop"
 }
 ```
 
-in the __main.tf__ file add a new variable for 'env_name'
+In the _main.tf_, file add a new variable called `env_name`:
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:372]: Again, reference to the location of _main.tf_ file can be helpful.

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 5]
```diff
@@ -413,11 +463,15 @@ inputs = {
 }
 ```
 
-Run `terragrunt plan`. Looking at the output the `env_name` variable will now have 'dev:' added. Run `terragrunt apply` to update your configurations.
+Adding a `locals` block creates a local variable. Using the replace built-in function removes the unwanted parts of the relative path. And updating the `inputs` block to use the local variable.
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:466]: Can we rephrase this? The language doesn't seems to convey the context properly. I suggest something like. 
```suggestion
Here, you added a `locals` block to create a local variable and used the built-in `replace` function to remove the unwanted parts of the relative path.  Then, you updated the `inputs` block to use the local variable.
```

File: src/markdown-pages/terraform/terragrunt-configuration.mdx [hunk 6]
```diff
@@ -436,94 +490,110 @@ Run `terragrunt plan`. Looking at the output the `env_name` variable will now ha
...
-<Callout variant="tip">
+</Step>
 
-To give access to the S3 bucket in your Amazon account, create an IAM user. Give the IAM user access to the S3 bucket storing the terraform state.
+<Step>
 
-</Callout>
+[Create an IAM user](https://docs.aws.amazon.com/IAM/latest/UserGuide/intro-structure.html) with [permissions to store files in your bucket](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_s3_rw-bucket.html).
+
+In your root _terragrunt.hcl_, add a `remote_state` block that tells Terragrunt where to place your file in S3:
```
Comment (mehreentahir16) [inline src/markdown-pages/terraform/terragrunt-configuration.mdx:532]: Shouldn't this be a separate step?

File: src/markdown-pages/terraform/terragrunt-configuration.mdx (hunk 5 above)
Comment (alexronquillo) [reply src/markdown-pages/terraform/terragrunt-configuration.mdx:466]: Agreed. I mostly copied this from above and changed the commas to periods. I should have edited the language. Good catch!

File: src/markdown-pages/terraform/terragrunt-configuration.mdx (hunk 6 above)
Comment (alexronquillo) [reply src/markdown-pages/terraform/terragrunt-configuration.mdx:532]: yep

**Review by mehreentahir16 — APPROVED** [review]

**Pull request merged**